js/testapps/format-tester/src/index.ts

```diff
@@ -0,0 +1,155 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { googleAI } from '@genkit-ai/googleai';
+import { vertexAI } from '@genkit-ai/vertexai';
+import { CallableFlow, GenerateOptions, genkit, z } from 'genkit';
+import { logger } from 'genkit/logging';
+
+logger.setLogLevel('debug');
+
+const ai = genkit({
+  plugins: [googleAI(), vertexAI()],
+});
+
+function formatFlow(name: string, options: GenerateOptions) {
+  return ai.defineFlow(
+    {
+      name,
+      inputSchema: z.string(),
+    },
+    async (model) => {
+      console.log('\n===', name, 'with model', model);
+      options.model = model;
+      options.config = {
+        safetySettings: [
+          {
+            category: 'HARM_CATEGORY_HATE_SPEECH',
+            threshold: 'BLOCK_NONE',
+          },
+          {
+            category: 'HARM_CATEGORY_DANGEROUS_CONTENT',
+            threshold: 'BLOCK_NONE',
+          },
+          {
+            category: 'HARM_CATEGORY_HARASSMENT',
+            threshold: 'BLOCK_NONE',
+          },
+          {
+            category: 'HARM_CATEGORY_SEXUALLY_EXPLICIT',
+            threshold: 'BLOCK_NONE',
+          },
+        ],
+      };
+      const { stream, response } = await ai.generateStream(options);
+      for await (const chunk of stream) {
+        console.log('text:', chunk.text);
+        console.log('output:', chunk.output);
+      }
+      console.log();
+      console.log('final output:', (await response).output);
+    }
+  );
+}
+
+const prompts: Record<string, GenerateOptions> = {
+  text: {
+    prompt: 'tell me a short story about pirates',
+    output: { format: 'text' },
+  },
+  json: {
+    prompt: 'generate a creature for an RPG game',
+    output: {
+      format: 'json',
+      schema: z.object({
+        name: z.string().describe('the name of the creature'),
+        backstory: z
+          .string()
+          .describe('a one-paragraph backstory for the creature'),
+        hitPoints: z.number(),
+        attacks: z.array(z.string()).describe('named attacks'),
+      }),
+    },
+  },
+  array: {
+    prompt: 'generate a list of characters from Futurama',
+    output: {
+      format: 'array',
+      schema: z.array(
+        z.object({
+          name: z.string(),
+          description: z.string(),
+          friends: z.array(z.string()),
+          enemies: z.array(z.string()),
+        })
+      ),
+    },
+  },
+  jsonl: {
+    prompt: 'generate fake products for an online pet store',
+    output: {
+      format: 'array',
+      schema: z.array(
+        z.object({
+          name: z.string(),
+          description: z.string(),
+          price: z.number(),
+          stock: z.number(),
+          color: z.string(),
+          tags: z.array(z.string()),
+        })
+      ),
+    },
+  },
+  enum: {
```

Comment:
what's the streaming expectation for `enum`? I'm seeing

```json
{"text":"HIGH \n","output":null}
```

in non-streaming mode I see `response.output` as `HIGH`.

so, should expect in streaming mode to see

```json
{"text":"HIGH \n","output":"HIGH"}
```

right?

Reply:
I didn't write a chunk parser for enum mode since it doesn't make sense to "partially" get an enum...maybe it should just throw an error instead of nulling?